Doc/library/unittest.mock.rst

```diff
@@ -1565,25 +1566,44 @@ Imagine we have a project that we want to test with the following structure::
         -> from a import SomeClass
         -> some_function instantiates SomeClass
 
-Now we want to test ``some_function`` but we want to mock out ``SomeClass`` using
-:func:`patch`. The problem is that when we import module b, which we will have to
-do then it imports ``SomeClass`` from module a. If we use :func:`patch` to mock out
-``a.SomeClass`` then it will have no effect on our test; module b already has a
-reference to the *real* ``SomeClass`` and it looks like our patching had no
-effect.
+Now we want to test ``some_function`` but we want to mock out ``SomeClass``
+using :func:`patch`. An incorrect way to write a test is as follows::
 
-The key is to patch out ``SomeClass`` where it is used (or where it is looked up).
-In this case ``some_function`` will actually look up ``SomeClass`` in module b,
-where we have imported it. The patching should look like::
+    import b
+
+    @patch('a.SomeClass')
+    def test1(MockClass):
+        -> configure the mock
+        assert b.some_function(...)
+
+The problem is that module b imports ``SomeClass`` from module a. Hence, within
+the context of the ``test`` function, the name ``a.SomeClass`` points to the
+``MockClass``, however module b already has a reference to the *real* (non-mock)
+``SomeClass``.
```

Comment:
I think the sentence `...then it will have no effect on our test` is important as it emphasizes that doing that has no effect (which makes concrete the assertion "an incorrect way to...").